### DISCUSSION TITLE
Multiplying points in V6

### DISCUSSION THREAD

**Discussion by milonic** [fabricjs/fabric.js#9709] [Q&A]
Hi Guys,

In V6, this returns {NaN,NaN}: **console.log(new fabric.Point(10, 10).multiply(3))**

What's the new way now that chaining has been removed?

**Comment by milonic** [comment]
It's OK I found it - I needed scalarMultiply

**Comment by milonic** [comment]
Just a suggestion but would changing:

```
    multiply(that) {
      return new Point(this.x * that.x, this.y * that.y);
    }

```

to

```
    multiply(that) {
      if(!that.x && !that.y)return this.scalarMultiply(that)
      return new Point(this.x * that.x, this.y * that.y);
    }
```

be a bad thing?

**Reply by ShaMan123** [reply]
Yes
We changed it to align the naming.
I know it is breaking but typescript should warn you about it
If you do not use it consider running it as a standalone command just to see you are ok

**Comment by asturur** [comment]
> Just a suggestion but would changing:
> 
> ```
>     multiply(that) {
>       return new Point(this.x * that.x, this.y * that.y);
>     }
> ```
> 
> to
> 
> ```
>     multiply(that) {
>       if(!that.x && !that.y)return this.scalarMultiply(that)
>       return new Point(this.x * that.x, this.y * that.y);
>     }
> ```
> 
> be a bad thing?

More than a bad thing would be confusing.
Now fabricJS should give you auto completition on every method and class and we are trying to move to a situation in which is clear what a function does and which side effects has.
In your example a point that is 0,0 would break for example.
And of course we could write code that detects that, but then every single multiplication will carry the load of that special edge case instead of being a direct multiplication.

I can understand that there are environments in which that can be useful, you can still add a function to the Point prototype to do that

```
Point.prototype.multiplyFlexlible(that) {
  if (that instanceof Point) return this.multiply(that);
  return this.scalarMultiply(that);
}
```

and just use it when you need it.
